lucene/core/src/java/org/apache/lucene/index/BufferedUpdates.java

```diff
@@ -288,15 +184,24 @@ void clear() {
     deleteTerms.clear();
     deleteQueries.clear();
     deleteDocIDs.clear();
-    numericUpdates.clear();
-    binaryUpdates.clear();
     numTermDeletes.set(0);
-    numNumericUpdates.set(0);
-    numBinaryUpdates.set(0);
-    bytesUsed.set(0);
+    numFieldUpdates.set(0);
+    fieldUpdates.clear();
+    bytesUsed.addAndGet(-bytesUsed.get());
+    fieldUpdatesBytesUsed.addAndGet(-fieldUpdatesBytesUsed.get());
   }
   
   boolean any() {
-    return deleteTerms.size() > 0 || deleteDocIDs.size() > 0 || deleteQueries.size() > 0 || numericUpdates.size() > 0 || binaryUpdates.size() > 0;
+    return deleteTerms.size() > 0 || deleteDocIDs.size() > 0 || deleteQueries.size() > 0 || numFieldUpdates.get() > 0;
+  }
+
+  @Override
+  public long ramBytesUsed() {
+    return bytesUsed.get() + fieldUpdatesBytesUsed.get();
+  }
+
+  void clearDeletedDocIds() {
+    deleteDocIDs.clear();
+    bytesUsed.addAndGet(-deleteDocIDs.size() * BufferedUpdates.BYTES_PER_DEL_DOCID);
   }
```

Comment:
The `byteUsed` will add nothing if we clear `deleteDocIDs` first, cuz its size is always zero.
Maybe a typo here?

Reply:
that's a bug indeed. It might not be a huge deal since we don't do much with the bytesUsed after this was called but it should still be fixed. Do you want to open a PR for this?

Reply:
I'm glad to do so😊